app/src/main/res/layout/fragment_user_login_credentials.xml

```diff
@@ -38,19 +38,42 @@
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@id/username" />
 
-    <EditText
-        android:id="@+id/password"
-        style="@style/Input.Default"
-        android:layout_width="0dp"
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:id="@+id/password_constraint_layout"
+        android:layout_width="match_parent"
         android:layout_height="wrap_content"
         android:layout_marginTop="8dp"
-        android:autofillHints="password"
-        android:imeOptions="actionDone"
-        android:inputType="textPassword"
-        android:privateImeOptions="horizontalAlignment=right"
-        app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@id/password_label" />
+        app:layout_constraintTop_toBottomOf="@id/password_label" >
+
+        <EditText
+            android:id="@+id/password"
+            style="@style/Input.Default"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+
+            android:autofillHints="password"
+            android:imeOptions="actionDone"
+            android:inputType="textPassword"
+            android:privateImeOptions="horizontalAlignment=right"
+            app:layout_constraintEnd_toStartOf="@+id/toggle_password_visibility_button"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintBottom_toBottomOf="@id/toggle_password_visibility_button"
+            />
+
+        <ImageButton
```

Comment:
The background is wrong when focused:

![image](https://github.com/user-attachments/assets/59ce0fb8-6490-463e-9bd1-9828f1f9c910)

The ripple effect on activation seems to be fine.